File: AssayDashboard/Source/Hologic.AssayDashboard.Client/src/app/assayviewrepo/assayviewrepo-models/assayflag.ts
/*
// ----------------------------------------------------------------
// Copyright (c) 2017 Hologic, Inc.  All Rights Reserved.
// ----------------------------------------------------------------
*/
export class AssayFlag {
    flagName: string;
    resultBit: String;
    flagId: String;
    flagSeverity: String;
    flagLegend: String;

    constructor() {
        this.flagName = undefined;
        this.resultBit = undefined;
        this.flagId = undefined;
        this.flagSeverity = undefined;
        this.flagLegend = undefined;
    }
}

File: AssayDashboard/Source/Hologic.AssayDashboard.AssayParser/Models/AssayFlag.cs
using System.Xml.Serialization;

namespace Hologic.AssayDashboard.AssayParser
{
    public class AssayFlag
    {
        [XmlAttribute("AssayFlagName")]
        public string flagName;

        [XmlAttribute("ReportedResultBit")]
        public string resultBit;

        [XmlAttribute("AssayFlagID")]
        public string flagId;

        [XmlAttribute("AssayFlagSeverity")]
        public string flagSeverity;

        [XmlElement("FlagLegend")]
        public string flagLegend;
    }
}